### PR TITLE
Add Search Page

### PR DESCRIPTION
**Pull request by conradbeach**
## Goal
Pivotal Tracker: [#171524287](https://www.pivotaltracker.com/story/show/171524287), [#171524862](https://www.pivotaltracker.com/story/show/171524862), [#171524339](https://www.pivotaltracker.com/story/show/171524339)

This adds an initial, rudimentary search page. It allows you to search for a student and displays the 
 first page of results. It's missing things such as the ability to page through other results and displaying all of the necessary user information.

I'd written enough code that I wanted to get it reviewed before going farther.

## Demo
[Search Page](https://pull-request-demo-files.s3-us-west-2.amazonaws.com/search_page.mp4)

### CODE REVIEW

**Comment by conradbeach** [comment]
After more testing, it looks like the Canvas API endpoint we're using includes users from child sub-accounts, so we won't have to implement that ourselves like I initially thought.

File: app/controllers/api/canvas_users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Api::CanvasUsersController < ApplicationController
```
Comment (bfcoder) [inline app/controllers/api/canvas_users_controller.rb]: We should probably add a validate token here, yes? If you inherit from our api application base class, you will get that validation:
`< Api::ApiApplicationController`

Comment (conradbeach) [reply app/controllers/api/canvas_users_controller.rb]: Added. Thanks for catching that.

**Comment by conradbeach** [comment]
@bfcoder Any other changes you'd like made for this? Or is it good to go?

**Comment by conradbeach** [comment]
Thanks

**Pull request merged**